Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -1114,4 +1222,143 @@ extension ApolloCodegenConfiguration {
       experimentalFeatures: experimentalFeatures,
       schemaDownloadConfiguration: schemaDownloadConfiguration)
   }
+
+  /// Enum to enable using
+  /// [Automatic Persisted Queries (APQs)](https://www.apollographql.com/docs/apollo-server/performance/apq)
+  /// with your generated operations.
+  ///
+  /// APQs are an Apollo Server feature. When using Apollo iOS to connect to any other GraphQL server,
+  /// `APQConfig` should be set to `.disabled`
+  public enum APQConfig: String, Decodable {
+    /// The default value. Disables APQs.
+    /// The operation document is sent to the server with each operation request.
+    @available(*, deprecated, message: "Use PersistedQueryConfig.disabled instead.")
+    case disabled
+
+    /// Automatically persists your operations using Apollo Server's
+    /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
+    @available(*, deprecated, message: "Use PersistedQueryConfig.automaticallyPersistedQueries instead.")
+    case automaticallyPersist
+
+    /// Provides only the `operationIdentifier` for operations that have been previously persisted
+    /// to an Apollo Server using
+    /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
+    ///
+    /// If the server does not recognize the `operationIdentifier`, the operation will fail. This
+    /// method should only be used if you are manually persisting your queries to an Apollo Server.
+    @available(*, deprecated, message: "Use PersistedQueryConfig.safelistedQueriesOnly instead.")
+    case persistedOperationsOnly
+
+    var operationDocumentFormat: ApolloCodegenConfiguration.OperationDocumentFormat {
+      switch self {
+      case .disabled:
+        return .definition
+      case .automaticallyPersist:
+        return [.definition, .operationId]
+      case .persistedOperationsOnly:
+        return .operationId
+      }
+    }
+  }
+}
+
+extension ApolloCodegenConfiguration.FileOutput {
+  /// Deprecated initializer.
+  ///
+  /// - Parameters:
+  ///  - schemaTypes: The local path structure for the generated schema types files.
+  ///  - operations: The local path structure for the generated operation object files.
+  ///  Defaults to `.inSchemaModule`.
+  ///  - testMocks: The local path structure for the test mock operation object files.
+  ///  If `.none`, test mocks will not be generated. Defaults to `.none`.
+  ///  - operationIdentifiersPath: An absolute location to an operation id JSON map file
+  ///  for use with APQ registration. Defaults to `nil`.
+  @available(*, deprecated, renamed: "init(schemaTypes:operations:testMocks:operationManifest:)")
+  @_disfavoredOverload
```

Comment:
I like this! Should we add it to the other deprecated initializer above?

Reply:
Actually I learned something that we can do to make this better. Instead of the `@disfavoredOverload`, if you remove the default value for the deprecated parameter, it will only use this overload if you are manually setting that parameter.

Reply:
And the other one above doesn't need this b/c its a changing of the name of a parameter, so the new init is not ever seen as an overload of the deprecated one.

Reply:
Good to know!